src/main/java/au/com/addstar/truehardcore/TrueHardcore.java

```diff
@@ -306,7 +308,7 @@ public void onEnable() {
         enableCombatLog(cfg.antiCombatLog);
         if (cfg.autoSaveEnabled) {
             log("Launching auto-save timer (every 5 minutes)...");
-            getServer().getScheduler().runTaskTimer(this, this::saveInGamePlayers,
+            getServer().getScheduler().runTaskTimer(this, this::runMaintenance,
```

Comment:
I think this is a great idea! However I don't understand some implementation choices.

Why did you add the chunk loading code into the `autoSaveEnabled` functionality? Which is always disabled by default because its very buggy. You should separate these two things since they have nothing to do with each other.

It's fine to have a "maintenance" timer but it shouldnt be hooked into the config flag for auto save. I don't mind if you move the auto save functionality into the new maintenance routine (with a condition on `cfg.autoSaveEnabled`) but these are two completely independent things.

My suggestion would be to have a 60s generic maintenance timer, then call this `expireOldChunks` each interval (so the max chunk load time is 6 minutes, not 10) and have the auto save inside this maintenance routine but only called if the config option is enabled (which it never is currently because its broken).

Reply:
Because I flicked in in there without reading the code properly